core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,356 @@
+          $scope.openAddSolutionModal();
+          return;
+        }
+        //$scope.isAVerifiedAnswer.$$state.value.outcome.dest
+        if (editorContextService.getActiveStateName() !== (
+          $scope.isAVerifiedAnswer.$$state.value.outcome.dest)) {
+          stateSolutionService.saveDisplayedValue();
+          $scope.solutionIsSpecified = (
+            JSON.stringify(stateSolutionService.savedMemento) ? true : false);
+        } else {
+          alertsService.addInfoMessage('That solution was incorrect!');
+          $scope.openAddSolutionModal();
+        }
+      });
+    };
+
+    $scope.deleteSolution = function(evt) {
+      // Prevent clicking on the delete button from also toggling the display
+      // state of the answer group.
+      evt.stopPropagation();
+
+      alertsService.clearWarnings();
+      $modal.open({
+        templateUrl: 'modals/deleteSolution',
+        backdrop: true,
+        controller: [
+          '$scope', '$modalInstance', function($scope, $modalInstance) {
+            $scope.reallyDelete = function() {
+              $modalInstance.close();
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function() {
+        stateSolutionService.displayed = {};
+        stateSolutionService.saveDisplayedValue();
+        $scope.solutionIsSpecified = false;
+      });
+    };
+
+    $scope.onComponentSave = function() {
+      alertsService.clearWarnings();
+      var explorationId =
+        explorationContextService.getExplorationId();
+      var currentStateName =
+        editorContextService.getActiveStateName();
+      var state = explorationStatesService.getState(currentStateName);
+      var interactionId = stateInteractionIdService.savedMemento;
+      var rulesServiceName =
+        angularNameService.getNameOfInteractionRulesService(
+          interactionId);
+      // Inject RulesService dynamically.
+      var rulesService = $injector.get(rulesServiceName);
+      var answer = stateSolutionService.savedMemento.correctAnswer;
+      if ($rootScope.objectType === 'RegionSelect') {
+        answer = {
+          clickPosition: [0, 0],
+          clickedRegions: [stateSolutionService.savedMemento.correctAnswer]
+        };
+      }
+
+      try {
+        var verifiedAnswer = (
+          AnswerClassificationService.getMatchingClassificationResult(
+            explorationId, state, answer, true, rulesService));
+      }
+      catch (e) {
+        alertsService.addInfoMessage('That solution was invalid!');
+        stateSolutionService.saveDisplayedValue();
+        $scope.solutionIsSpecified = (
+          JSON.stringify(stateSolutionService.displayed) ? true : false);
+        return;
+      }
+      if (editorContextService.getActiveStateName() !== (
+        verifiedAnswer.$$state.value.outcome.dest)) {
```

Comment:
Hm, this doesn't look great. Can you do this without referencing obviously-private properties like $$state?